libs/seo/guides/state.md

```diff
@@ -0,0 +1,44 @@
+# `@daffodil/seo/state`
+
+## Angular Router Page Hooks
+
+Applications have the following SEO requirements with regards to navigation:
+- Adding SEO data to the page in response to arbitrary redux actions during navigation.
+- Removing added SEO data when navigating to new pages.
+- Preserving added SEO data when navigation fails.
+
+In addition to the services in `@daffodil/seo`, `@daffodil/seo/state` provides mechanisms for hooking into the Angular router navigation flow with respect to SEO updates. Application features can define behavior for getting SEO data in response to particular state actions by providing to SEO feature-specific injection tokens.
+
+### Example
+
+The following example demonstrates adding canonical URLs on a category page in response to the category page load success action.
+
+```ts
+@NgModule({
+  imports: [
+    DaffSeoStateModule,
+    DaffCategoryStateModule,
```

Comment:
yes, I didn't think it was quite in the scope of this example. However I think its best to just create a stackblitz to this effect. This will be a draft until then.